src/modeler/AutoRest.Swagger/SwaggerParser.cs

```diff
@@ -22,14 +23,19 @@ public static ServiceDefinition Load(string path, IFileSystem fileSystem)
                 throw new ArgumentNullException("fileSystem");
             }
 
-            return SwaggerParser.Parse(fileSystem.ReadFileAsText(path));
+            var swaggerDocument = fileSystem.ReadFileAsText(path);
+            if (path.EndsWith(".md"))
```

Comment:
done! funny enough, I caught a bug in a test case with that: `src/dev/TestServer/swagger/azure-special-properties.json` contained *duplicate* `required` keys, which made it invalid YAML/JSON... so it was parsed as markdown, which failed.
Sadly, the `JsonDeserializer` swallowed that bug so far, now it was exposed by the explicit check.